GVFS/GVFS.Common/Http/HttpRequestor.cs

```diff
@@ -31,22 +42,51 @@ static HttpRequestor()
             availableConnections = new SemaphoreSlim(ServicePointManager.DefaultConnectionLimit);
         }
 
-        public HttpRequestor(ITracer tracer, RetryConfig retryConfig, GitAuthentication authentication)
+        protected HttpRequestor(ITracer tracer, RetryConfig retryConfig, Enlistment enlistment)
         {
-            this.client = new HttpClient(new HttpClientHandler() { UseDefaultCredentials = true });
-            this.client.Timeout = retryConfig.Timeout;
             this.RetryConfig = retryConfig;
-            this.authentication = authentication;
+
+            this.authentication = enlistment.Authentication;
 
             this.Tracer = tracer;
 
+            var httpClientHandler = new HttpClientHandler() { UseDefaultCredentials = true };
+#if DEBUG

```

Comment:
I don't think the other reviews have called this out yet. Please avoid precompiler constants - you'll notice that we don't have a single one in any of our C# files.

Two other techniques that you can use: 
* git config setting that enables/disables self-signed certs
* Call `ProcessHelper.IsDevelopmentVersion`. This is dangerous though, because we don't want to ship different code than what we test, so I would personally favor a bool config setting that just says yes/no to the question of self-signed certs

I would also encapsulate all of the logic related to self-signed certs into one place. Right now I see a sprinkling of these `#if DEBUG` checks, and that is going to be difficult to maintain.

Reply:
Git actually already has http.sslVerify setting, which I'll reuse for this.